server/src/main/java/org/elasticsearch/transport/SniffConnectionStrategy.java

```diff
@@ -299,31 +301,53 @@ private void collectRemoteNodes(Iterator<Supplier<DiscoveryNode>> seedNodesSuppl
                 }
                 final Transport.Connection connection = openConnectionStep.result();
 
-                ClusterStateRequest request = new ClusterStateRequest();
-                request.clear();
-                request.nodes(true);
                 // here we pass on the connection since we can only close it once the sendRequest returns otherwise
                 // due to the async nature (it will return before it's actually sent) this can cause the request to fail
                 // due to an already closed connection.
                 ThreadPool threadPool = transportService.getThreadPool();
                 ThreadContext threadContext = threadPool.getThreadContext();
-                TransportService.ContextRestoreResponseHandler<ClusterStateResponse> responseHandler =
-                    new TransportService.ContextRestoreResponseHandler<>(
-                        threadContext.newRestorableContext(false),
-                        new SniffClusterStateResponseHandler(connection, listener, seedNodesSuppliers)
-                    );
-                try (ThreadContext.StoredContext ignore = threadContext.stashContext()) {
-                    // we stash any context here since this is an internal execution and should not leak any
-                    // existing context information.
-                    threadContext.markAsSystemContext();
-                    transportService.sendRequest(
-                        connection,
-                        ClusterStateAction.NAME,
-                        request,
-                        TransportRequestOptions.EMPTY,
-                        responseHandler
-                    );
+
+                if (REMOTE_CLUSTER_PROFILE.equals(connectionManager.getConnectionProfile().getTransportProfile())) {
```

Comment:
I prefer to keep it as is because there is a subtle implication in moving the logic: The `_remote_cluster` profile is special and meaningful only when it is accessed inside the area of new additions to remote cluster code. The profile name itself is not reserved. Theoretically (though it won't happen today) we cannot just rely on the name itself to determine whether the profile is for remote cluster. It needs the usage context. I also wanted to keep the generic `ConnectionProfile` class unware of any remote cluster logic. Also given there are only two reptitions of this check, I think it's OK to have them as is.